src/proj7BittingCerratoCohenEllmer/bantam/lexer/Scanner.java

```diff
@@ -260,9 +234,15 @@ private boolean isCompleteInt(Stack<Character> spellingStack) {
     private boolean isCompleteString(Stack<Character> spellingStack) {
         int stackSize = spellingStack.size();
         if (stackSize > 1 && stackSize <= 5000) {
-            return spellingStack.peek() == '"' && spellingStack.get(stackSize - 1) != '\\';
+            if (spellingStack.peek() == '"' && spellingStack.get(stackSize - 2) != '\\') {
+                skippedLastToken = '\0';
+                return true;
+            } else {
+                return false;
+            }
         } else if (stackSize > 5000) {
             //raise error;
+            skippedLastToken = '\0';
```

Comment:
Im not sure if this is the correct way to think about it, but it might be good to not cut off the string at 5000 characters. I think we should accept all incoming chars ( that are between "") in the case of strings. Maybe throw in a call to a void "checkStringErrors()" method that registers string length and illegal symbol errors? This would also need to keep track of the line number the first quote appears on for multiline errors. I can start this on the next branch if thats good

Reply:
Looking at the instructions I am more confident in saying illegal string tokens should be fully scanned and registered as an error token. This will also prevent ruining the next token and make it possible to find unterminated strings.

Reply:
String constants cannot span multiple lines in Bantam java though I agree with the premature ending for sure. Illegal escape characters is \ followed by anything but n,t,",\,f.